e2e-tests/functions

```diff
@@ -466,3 +469,95 @@ check_passwords_leak() {
 		collect_logs $OPERATOR_NS
 	fi
 }
+
+calculate_backup_destination() {
+	local cluster=$1
+	local backup_name=$2
+
+	local repo
+	local storage_type
+
+	repo=$(kubectl get pg-backup -n "$NAMESPACE" "$backup_name" -o yaml \
+		| yq ".spec.repoName")
+
+	if [[ $(kubectl get pg -n "$NAMESPACE" "$cluster" -o yaml \
+		| yq ".spec.backups.pgbackrest.repos[] | select(.name==\"$repo\") | has(\"s3\")") == "true" ]]; then
+		storage_type="s3"
+	elif [[ $(kubectl get pg -n "$NAMESPACE" "$cluster" -o yaml \
+		| yq ".spec.backups.pgbackrest.repos[] | select(.name==\"$repo\") | has(\"gcs\")") == "true" ]]; then
+		storage_type="gcs"
+	elif [[ $(kubectl get pg -n "$NAMESPACE" "$cluster" -o yaml \
+		| yq ".spec.backups.pgbackrest.repos[] | select(.name==\"$repo\") | has(\"azure\")") == "true" ]]; then
+		storage_type="azure"
+	else
+		echo "ERROR: unknown storage type"
+		exit 1
+	fi
+
+	local repo_path
+	local bucket
+
+	repo_path=$(kubectl get pg -n "$NAMESPACE" "$cluster" -o yaml \
+		| yq ".spec.backups.pgbackrest.global.$repo-path")
+	bucket=$(kubectl get pg -n "$NAMESPACE" "$cluster" -o yaml \
+		| yq ".spec.backups.pgbackrest.repos[] | select(.name==\"$repo\").$storage_type.bucket")
+
+	if [[ $storage_type == "gcs" ]]; then
+		storage_type="gs"
+	fi
+
+	echo -n "$storage_type://$bucket$repo_path"
+}
+
+get_backup_name_by_job() {
+	local job_name=$1
+
+	pgbackup_per_job=$(kubectl get pg-backup -n "$NAMESPACE" -o yaml \
+		| yq "[.items[] | select(.status.jobName==\"$job_name\")]" \
+		| yq '. | length')
+	if [[ $pgbackup_per_job != 1 ]]; then
+		echo "ERROR: pgbackup_per_job != 1"
+		exit 1
+	fi
+
+	kubectl get pg-backup -n "$NAMESPACE" -o yaml \
+		| yq ".items[] | select(.status.jobName==\"$job_name\").metadata.name"
+}
+
+check_jobs_and_pgbackups() {
+	local cluster=$1
+
+	readarray -t job_names <<<"$(kubectl get job -n "$NAMESPACE" -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')"
```

Comment:
on Mac OS
`functions: line 530: readarray: command not found`

Reply:
https://github.com/percona/percona-postgresql-operator/pull/524/commits/12e320d018130aea6ff3789bc62df7705c53f8c5